app/src/main/java/com/github/se/gomeet/ui/mainscreens/events/Events.kt

```diff
@@ -187,81 +185,110 @@ fun Events(nav: NavigationActions, userViewModel: UserViewModel, eventViewModel:
+                } else {
+                  Column(
+                      modifier = Modifier.verticalScroll(rememberScrollState()).fillMaxSize(),
+                      horizontalAlignment = Alignment.CenterHorizontally) {
+                        if (selectedFilter == ALL || selectedFilter == JOINED) {
+                          if (joinedEvents.isNotEmpty()) {
+                            Spacer(modifier = Modifier.height(screenHeight / 40))
+                            Text(
+                                text = JOINED.formattedName,
+                                style = MaterialTheme.typography.titleLarge,
+                                color = MaterialTheme.colorScheme.onBackground,
+                                modifier =
+                                    Modifier.padding(horizontal = screenWidth / 15)
+                                        .testTag("JoinedTitle"))
+                            joinedEvents.forEach { event ->
                               ShowWidgets(
                                   event = event,
                                   query = query.value,
                                   nav = nav,
                                   userVM = userViewModel)
                             }
-                      }
-
-                      // Display favourite events if 'All' or 'Favourites' is selected
-                      if (selectedFilter == ALL || selectedFilter == FAVOURITES) {
-                        Spacer(modifier = Modifier.height(screenHeight / 40))
-                        Text(
-                            text = FAVOURITES.formattedName,
-                            style = MaterialTheme.typography.titleLarge,
-                            color = MaterialTheme.colorScheme.onBackground,
-                            modifier =
-                                Modifier.padding(horizontal = screenWidth / 15)
-                                    .testTag("FavouritesTitle"))
+                          }
+                        }
 
-                        // Loop through and display events are marked favourites by the currentUser
-                        eventList
-                            .filter { e -> user.value!!.myFavorites.contains(e.eventID) }
-                            .forEach { event ->
+                        if (selectedFilter == ALL || selectedFilter == FAVOURITES) {
+                          if (favouriteEvents.isNotEmpty()) {
+                            Spacer(modifier = Modifier.height(screenHeight / 40))
+                            Text(
+                                text = FAVOURITES.formattedName,
+                                style = MaterialTheme.typography.titleLarge,
+                                color = MaterialTheme.colorScheme.onBackground,
+                                modifier =
+                                    Modifier.padding(horizontal = screenWidth / 15)
+                                        .testTag("FavouritesTitle"))
```

Comment:
This testTag shouldn't be left as a hardcoded string - put it in the resource manager in values/strings.xml and fetch it using getString(LocalContext.current, R.string.string_name)